cnd/tests/ethereum_transaction_pattern_e2e.rs

```diff
@@ -12,6 +12,18 @@ use web3::{
     Web3,
 };
 
+pub fn new_web3_client<D: Docker, E: Image>(
```

Comment:
It's a small optimisation to follow the common test conventions: set-up, execute, assert.

Reply:
Not disputing the change but wanted to clarify that usually this pattern only refers to the code inside the test function, not to the order of fns inside the file :)

Reply:
Happy to drop the commit. It is an artefact from the time when I tried to test this via rust.

Reply:
I like to think about the structure of source code similarly to that of a book.

The first thing you read (ToC) should give you and overview about the software: directory and file names.
With a file (chapter / section), things make sense if you read from top to bottom.
You probably want to start with what is most important and keep refining the story you are telling.
Mapped to source code, that would be the most important function at the top of the file with functions that are being used in there beneath it in the order they are used.

That is why I originally placed this fn at the bottom: it is an implementation detail for the test (which is the most important thing in this file) and you probably don't even care about the details when you read over the test fn so you don't even have to scroll down.

If you agree, drop the commit, if you disagree, leave it :)

Reply:
Interestingly, `use` statements are at the top of the file and literally that last thing you are usually interested in :joy:

Reply:
> Interestingly, `use` statements are at the top of the file and literally that last thing you are usually interested in

That might be because 'every other language ever does this so just to not be too crazy lets just do it to'.  Rubbish reason if it is, I believe they are only at the top by convention, they can actually be placed anywhere.  Totally happy to start putting them at the bottom :)